lib/mix/lib/mix/tasks/test.ex

```diff
@@ -40,6 +40,8 @@ defmodule Mix.Tasks.Test do
       # Each line is only considered once.
       #
       # We use ETS for performance, to avoid working with nested maps.
+      results = discard_protocol_definitions(results)
```

Comment:
This _could_ be piped into `Enum.each` at the very beginning of the `try` block - I felt like keeping it outside of the `try` and table interactions was best. We could also filter before passing the results in here, as `:cover.modules()` is passed directly I was hesitant  but we could also do it right there.

Reply:
Thanks @PragTob! However, I am afraid ignoring protocols is going to hide the root cause. What if someone implements something like a protocol that does not have any code either?